src/libcore/fmt/mod.rs

```diff
@@ -265,7 +265,15 @@ pub struct ArgumentV1<'a> {
 // containing data (as a matter of static generation of the formatting
 // arguments), so this is merely an additional check.
 #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
-static USIZE_MARKER: fn(&usize, &mut Formatter<'_>) -> Result = |_, _| loop {};
+static USIZE_MARKER: fn(&usize, &mut Formatter<'_>) -> Result = |ptr, _| {
+    // This tries to ensure we don't collapse into some other function; we can't
+    // guarantee that doesn't happen, but this tries to make sure that if we
+    // *do* get collapsed with some other function, that function also takes a
+    // `usize` as it's first argument, ensuring correctness below when we read
+    // out the associated value from `ArgumentV1`.
+    let _v: usize = *crate::hint::black_box(ptr);
+    loop {}
```

Comment:
If you make this return `Ok(())`, you could actually call it before reading the `usize` from `Argument`, ensuring that what this closure does is guaranteed to happen, giving more confidence that if there was UB, it was invoked by calling the `fn` pointer which should be safe to call.

Reply:
It's a bit annoying to call the formatter as we don't have a `&mut Formatter<'_>` around and I'd rather not manufacture one just for this. I think we could likely thread it through but I think that hurts readability and such so I'd rather not.

Reply:
Oh, right, that method doesn't take the formatter, fair enough.